justfile

```diff
@@ -29,40 +31,40 @@ install-operator:
 
 # Generate the fixture files for the Celestia Mocha testnet using the prover parameter.
 # The prover parameter should be one of: ["mock", "network", "local"]
 # This generates the fixtures for all programs in parallel using GNU parallel.
 # If prover is set to network, this command requires the `SP1_PRIVATE_KEY` environment variable to be set.
-fixtures prover:
-  @echo "Generating fixtures for the Celestia Mocha testnet"
-  @echo "Building the operator..."
-  just build-operator
+fixtures prover: build-operator
   @echo "Generating fixtures... This may take a while (up to 20 minutes)"
+  TENDERMINT_RPC_URL="${TENDERMINT_RPC_URL%/}" && \
+  CURRENT_HEIGHT=$(curl "$TENDERMINT_RPC_URL"/block | jq -r ".result.block.header.height") && \
+  TRUSTED_HEIGHT=$(($CURRENT_HEIGHT-100)) && \
+  TARGET_HEIGHT=$(($CURRENT_HEIGHT-10)) && \
+  echo "For celestia fixtures, trusted block: $TRUSTED_HEIGHT, target block: $TARGET_HEIGHT, from $TENDERMINT_RPC_URL" && \
   parallel --progress --shebang --ungroup -j 4 ::: \
-    "RUST_LOG=info SP1_PROVER={{prover}} TENDERMINT_RPC_URL='https://rpc.celestia-mocha.com/' ./target/release/operator fixtures update-client --trusted-block 2438000 --target-block 2438010 -o 'contracts/fixtures/update_client_fixture.json'" \
-    "sleep 15 && RUST_LOG=info SP1_PROVER={{prover}} TENDERMINT_RPC_URL='https://rpc.celestia-mocha.com/' ./target/release/operator fixtures update-client-and-membership --key-paths clients/07-tendermint-0/clientState,clients/07-tendermint-001/clientState --trusted-block 2438000 --target-block 2438010 -o 'contracts/fixtures/uc_and_memberships_fixture.json'" \
-    "sleep 30 && RUST_LOG=info SP1_PROVER={{prover}} TENDERMINT_RPC_URL='https://rpc.celestia-mocha.com/' ./target/release/operator fixtures membership --key-paths clients/07-tendermint-0/clientState,clients/07-tendermint-001/clientState --trusted-block 2438000 -o 'contracts/fixtures/memberships_fixture.json'"
+    "RUST_LOG=info SP1_PROVER={{prover}} ./target/release/operator fixtures update-client --trusted-block $TRUSTED_HEIGHT --target-block $TARGET_HEIGHT -o 'contracts/fixtures/update_client_fixture.json'" \
+    "sleep 15 && RUST_LOG=info SP1_PROVER={{prover}} ./target/release/operator fixtures update-client-and-membership --key-paths clients/07-tendermint-0/clientState,clients/07-tendermint-001/clientState --trusted-block $TRUSTED_HEIGHT --target-block $TARGET_HEIGHT -o 'contracts/fixtures/uc_and_memberships_fixture.json'" \
+    "sleep 30 && RUST_LOG=info SP1_PROVER={{prover}} ./target/release/operator fixtures membership --key-paths clients/07-tendermint-0/clientState,clients/07-tendermint-001/clientState --trusted-block $TRUSTED_HEIGHT -o 'contracts/fixtures/memberships_fixture.json'"
+  cd e2e/interchaintestv8 && RUST_LOG=info SP1_PROVER=network GENERATE_FIXTURES=true go test -v -run '^TestWithSP1ICS07TendermintTestSuite/TestMisbehaviour$' -timeout 40m
```

Comment:
I ran into some issues when trying to run this one in parallel, but it is quite fast, so not as big of a deal.